tutors3/patches/openedx-common-settings

```diff
@@ -1,8 +1,8 @@
 DEFAULT_FILE_STORAGE = "storages.backends.s3boto3.S3Boto3Storage"
 VIDEO_IMAGE_SETTINGS["STORAGE_KWARGS"]["location"] = VIDEO_IMAGE_SETTINGS["STORAGE_KWARGS"]["location"].lstrip("/")
 VIDEO_TRANSCRIPTS_SETTINGS["STORAGE_KWARGS"]["location"] = VIDEO_TRANSCRIPTS_SETTINGS["STORAGE_KWARGS"]["location"].lstrip("/")
-GRADES_DOWNLOAD["STORAGE_KWARGS"]["location"] = GRADES_DOWNLOAD["STORAGE_KWARGS"]["location"].lstrip("/")
-GRADES_DOWNLOAD["STORAGE_KWARGS"]["bucket"] = "{{ S3_GRADE_BUCKET }}"
+GRADES_DOWNLOAD["STORAGE_KWARGS"] = {"location": GRADES_DOWNLOAD["STORAGE_KWARGS"]["location"].lstrip("/")}
```

Comment:
I presume you are overriding the whole dictionary here, rather than just overriding the value of `location`, in order to remove the problematic `base_url` key. Is that correct?

If so, I'm curious way why the `tutor-minio` plugin does not fail in the same way. Because it appears to be doing the same thing that we do here, namely only override the `location` key rather than the whole dictionary:

https://github.com/overhangio/tutor-minio/blob/ac817717c92ca006e7752efd1c034a6175f3c948/tutorminio/patches/openedx-common-settings

Reply:
>I presume you are overriding the whole dictionary here, rather than just overriding the value of location, in order to remove the problematic base_url key. Is that correct?

Exactly.

>If so, I'm curious way why the tutor-minio plugin does not fail in the same way. Because it appears to be doing the same thing that we do here, namely only override the location key rather than the whole dictionary:

I am not really familiar with the `tutor-minior` plugin. But please note that this problem occurs in Quince (3 months-ish ago?) and the last commit from that file dates back to 7 months ago. So they might have the same issue. I am not sure.

Reply:
No problem. @regisb, just checking, does this ring a bell? 

I ask because if setting `base_url` breaks both `tutor-minio` **and** `tutor-contrib-s3`, then it should probably be fixed in both — or else, maybe we want to remove this line from Tutor Core:

https://github.com/overhangio/tutor/blob/2924b92ceab8aca550ec5ecae559a367fb1b713a/tutor/templates/apps/openedx/settings/partials/common_all.py#L122

What do you think?

Reply:
Interesting. Thanks for the heads up Florian. @FahadKhalid210 could you please have a look and see if you can reproduce the issue described [here](https://github.com/hastexo/tutor-contrib-s3/pull/70)?

Reply:
@regisb sure

Reply:
@regisb The issue has been replicated locally by initiating Tutor with the `Quince` version and the `s3` plugin. Additionally, the problem was verified using `Minio`, and it was reproduced in that environment as well.

Reply:
Thanks @FahadKhalid210!

@regisb, at your convenience could you please advise whether the `base_url` setting is even still necessary in Tutor Core? Because if it is not, then I'd suggest we remove it from there, and remove any reference to it from the `s3` and `minio` plugins.

Reply:
I can't figure remember the reason I initially introduced a `base_url` setting in `GRADES_DOWNLOAD`. The most information I can find is from this commit, five years ago: https://github.com/overhangio/tutor/commit/abfbf1cc213afc97eb5bbc9452a3fe42cded5a77
Indeed, we should remove this `base_url` attribute if it's causing errors.
@FahadKhalid210 can you please handle this? We should verify that grades download keeps working both with and wihout minio.

Reply:
@FahadKhalid210: When you create a PR for such a change in either `tutor` or `tutor-minio`, would you mind adding a link to *this* issue in the discussion thread, so we get a backlink here and can coordinate the fix?

Reply:
Sure thing. Here it is: https://github.com/overhangio/tutor/pull/999